FRR is not compiled on FreeBSD with support for snmp

My configuration options:
```
./configure \
    --sysconfdir=/usr/local/etc/frr \
    --enable-pkgsrcrcdir=/usr/pkg/share/examples/rc.d \
    --localstatedir=/var/run/frr \
    --prefix=/usr/local \
    --enable-ospfclient=yes \
    --enable-ospfapi=yes \
    --enable-multipath=64 \
    --enable-user=frr \
    --enable-group=frr \
    --enable-vty-group=frrvty \
    --enable-configfile-mask=0640 \
    --enable-logfile-mask=0640 \
    --enable-rtadv \
    --enable-fpm \
    --with-pkg-git-version \
    --with-pkg-extra-version=-frr-develop \
   --enable-snmp=agentx
```
configuration status::
```
FRRouting configuration
------------------------------
FRR version             : 3.1-dev-frr-develop
host operating system   : freebsd11.1
source code location    : .
compiler                : gcc -std=gnu11
compiler flags          :  -g -Os -fno-omit-frame-pointer -funwind-tables -Wall -Wextra -Wmissing-prototypes -Wmissing-declarations -Wpointer-arith -Wbad-function-cast -Wwrite-strings -Wno-unused-result -Wno-unused-parameter -Wno-missing-field-initializers -D_THREAD_SAFE -pthread
make                    : gmake
linker flags            : -L/usr/local/lib -rdynamic -lexecinfo -lm -lcrypt   -ljson-c -lrt  -lreadline  -ltermcap -lm
state file directory    : /var/run/frr
config file directory   : /usr/local/etc/frr
example directory       : /usr/local/etc/frr
module directory        : /usr/local/lib/frr/modules
user to run as          : frr
group to run as         : frr
group for vty sockets   : frrvty
config file mask        : 0640
log file mask           : 0640
zebra protobuf enabled  : no
```
gmake result:
```
git suffix: -g215f9a9
git branches: local:master, github/FRRouting/frr.git/master
Files lib/gitversion.h.tmp and lib/gitversion.h are identical
gmake  all-recursive
gmake[1]: Entering directory '/var/www/develop/frr'
Making all in .
gmake[2]: Entering directory '/var/www/develop/frr'
git suffix: -g215f9a9
git branches: local:master, github/FRRouting/frr.git/master
Files lib/gitversion.h.tmp and lib/gitversion.h are identical
  CC       ldpd/ldpd.o
  AR       ldpd/libldp.a
  CC       lib/command.lo
  CC       lib/libfrr.lo
  CC       lib/module.lo
  CC       lib/pid_output.lo
  CC       lib/vty.lo
  CCLD     lib/libfrr.la
  CC       lib/lib_libfrrsnmp_la-agentx.lo
  CCLD     lib/libfrrsnmp.la
  CCLD     lib/libfrrzmq.la
  CC       ospfclient/ospf_apiclient.lo
  CCLD     ospfclient/libfrrospfapiclient.la
  CC       zebra/zebra_zebra_snmp_la-zebra_snmp.lo
In file included from ./lib/if.h:28:0,
                 from zebra/zebra_snmp.c:30:
./lib/hook.h:171:57: error: expected declaration specifiers or '...' before ')' token
 #define HOOK_ADDDEF(...) (void *hookarg , ## __VA_ARGS__)
                                                         ^
./lib/hook.h:189:17: note: in expansion of macro 'HOOK_ADDDEF'
   int(*funcptr) HOOK_ADDDEF arglist)                             \
                 ^~~~~~~~~~~
./lib/hook.h:194:2: note: in expansion of macro 'DECLARE_HOOK'
  DECLARE_HOOK(hookname, arglist, passlist)
  ^~~~~~~~~~~~
./lib/libfrr.h:110:1: note: in expansion of macro 'DECLARE_KOOH'
 DECLARE_KOOH(frr_early_fini, (), ())
 ^~~~~~~~~~~~
./lib/hook.h:171:57: error: expected declaration specifiers or '...' before ')' token
 #define HOOK_ADDDEF(...) (void *hookarg , ## __VA_ARGS__)
                                                         ^
./lib/hook.h:189:17: note: in expansion of macro 'HOOK_ADDDEF'
   int(*funcptr) HOOK_ADDDEF arglist)                             \
                 ^~~~~~~~~~~
./lib/hook.h:194:2: note: in expansion of macro 'DECLARE_HOOK'
  DECLARE_HOOK(hookname, arglist, passlist)
  ^~~~~~~~~~~~
./lib/libfrr.h:113:1: note: in expansion of macro 'DECLARE_KOOH'
 DECLARE_KOOH(frr_fini, (), ())
 ^~~~~~~~~~~~
In file included from zebra/zebra_snmp.c:25:0:
zebra/zebra_snmp.c: In function 'get_fwtable_route_node':
./lib/zebra.h:359:3: warning: implicit declaration of function 'typeof' [-Wimplicit-function-declaration]
   typeof(a) _a = (a);                                            \
   ^
zebra/zebra_snmp.c:346:8: note: in expansion of macro 'MIN'
        MIN(4U, *objid_len - v->namelen), &dest);
        ^~~
./lib/zebra.h:359:13: error: expected ';' before '_a'
   typeof(a) _a = (a);                                            \
             ^
zebra/zebra_snmp.c:346:8: note: in expansion of macro 'MIN'
        MIN(4U, *objid_len - v->namelen), &dest);
        ^~~
./lib/zebra.h:360:13: error: expected ';' before '_b'
   typeof(b) _b = (b);                                            \
             ^
zebra/zebra_snmp.c:346:8: note: in expansion of macro 'MIN'
        MIN(4U, *objid_len - v->namelen), &dest);
        ^~~
./lib/zebra.h:361:3: error: '_a' undeclared (first use in this function)
   _a < _b ? _a : _b;                                             \
   ^
zebra/zebra_snmp.c:346:8: note: in expansion of macro 'MIN'
        MIN(4U, *objid_len - v->namelen), &dest);
        ^~~
./lib/zebra.h:361:3: note: each undeclared identifier is reported only once for each function it appears in
   _a < _b ? _a : _b;                                             \
   ^
zebra/zebra_snmp.c:346:8: note: in expansion of macro 'MIN'
        MIN(4U, *objid_len - v->namelen), &dest);
        ^~~
./lib/zebra.h:361:8: error: '_b' undeclared (first use in this function)
   _a < _b ? _a : _b;                                             \
        ^
zebra/zebra_snmp.c:346:8: note: in expansion of macro 'MIN'
        MIN(4U, *objid_len - v->namelen), &dest);
        ^~~
./lib/zebra.h:359:13: error: expected ';' before '_a'
   typeof(a) _a = (a);                                            \
             ^
zebra/zebra_snmp.c:356:8: note: in expansion of macro 'MIN'
        MIN(4U, *objid_len - v->namelen - 6), &nexthop);
        ^~~
./lib/zebra.h:360:13: error: expected ';' before '_b'
   typeof(b) _b = (b);                                            \
             ^
zebra/zebra_snmp.c:356:8: note: in expansion of macro 'MIN'
        MIN(4U, *objid_len - v->namelen - 6), &nexthop);
        ^~~
gmake[2]: *** [Makefile:3674: zebra/zebra_zebra_snmp_la-zebra_snmp.lo] Error 1
gmake[2]: Leaving directory '/var/www/develop/frr'
gmake[1]: *** [Makefile:4232: all-recursive] Error 1
gmake[1]: Leaving directory '/var/www/develop/frr'
gmake: *** [Makefile:1870: all] Error 2
```
The screenshot looks more informative and beautiful:
![default](https://user-images.githubusercontent.com/11805503/34771971-1540930a-f60f-11e7-94d8-1fb842672eaa.png)

--- a/bgpd/Makefile.am
+++ b/bgpd/Makefile.am
@@ -116,7 +116,7 @@ module_LTLIBRARIES += bgpd_snmp.la
 endif
 
 bgpd_snmp_la_SOURCES = bgp_snmp.c
-bgpd_snmp_la_CFLAGS = $(WERROR) $(SNMP_CFLAGS)
+bgpd_snmp_la_CFLAGS = $(WERROR) $(SNMP_CFLAGS) -std=gnu99
 bgpd_snmp_la_LDFLAGS = -avoid-version -module -shared -export-dynamic
 bgpd_snmp_la_LIBADD = ../lib/libfrrsnmp.la
 

--- a/lib/subdir.am
+++ b/lib/subdir.am
@@ -190,7 +190,7 @@ if SNMP
 lib_LTLIBRARIES += lib/libfrrsnmp.la
 endif
 
-lib_libfrrsnmp_la_CFLAGS = $(WERROR) $(SNMP_CFLAGS)
+lib_libfrrsnmp_la_CFLAGS = $(WERROR) $(SNMP_CFLAGS) -std=gnu99
 lib_libfrrsnmp_la_LDFLAGS = -version-info 0:0:0
 lib_libfrrsnmp_la_LIBADD = lib/libfrr.la $(SNMP_LIBS)
 lib_libfrrsnmp_la_SOURCES = \

--- a/ospf6d/subdir.am
+++ b/ospf6d/subdir.am
@@ -61,6 +61,6 @@ ospf6d_ospf6d_SOURCES = \
 	# end
 
 ospf6d_ospf6d_snmp_la_SOURCES = ospf6d/ospf6_snmp.c
-ospf6d_ospf6d_snmp_la_CFLAGS = $(WERROR) $(SNMP_CFLAGS)
+ospf6d_ospf6d_snmp_la_CFLAGS = $(WERROR) $(SNMP_CFLAGS) -std=gnu99
 ospf6d_ospf6d_snmp_la_LDFLAGS = -avoid-version -module -shared -export-dynamic
 ospf6d_ospf6d_snmp_la_LIBADD = lib/libfrrsnmp.la

--- a/ospfd/subdir.am
+++ b/ospfd/subdir.am
@@ -89,7 +89,7 @@ ospfd_ospfd_LDADD = ospfd/libfrrospf.a lib/libfrr.la @LIBCAP@ @LIBM@
 ospfd_ospfd_SOURCES = ospfd/ospf_main.c
 
 ospfd_ospfd_snmp_la_SOURCES = ospfd/ospf_snmp.c
-ospfd_ospfd_snmp_la_CFLAGS = $(WERROR) $(SNMP_CFLAGS)
+ospfd_ospfd_snmp_la_CFLAGS = $(WERROR) $(SNMP_CFLAGS) -std=gnu99
 ospfd_ospfd_snmp_la_LDFLAGS = -avoid-version -module -shared -export-dynamic
 ospfd_ospfd_snmp_la_LIBADD = lib/libfrrsnmp.la
 

--- a/ripd/subdir.am
+++ b/ripd/subdir.am
@@ -35,7 +35,7 @@ ripd_ripd_SOURCES = \
 	# end
 
 ripd_ripd_snmp_la_SOURCES = ripd/rip_snmp.c
-ripd_ripd_snmp_la_CFLAGS = $(WERROR) $(SNMP_CFLAGS)
+ripd_ripd_snmp_la_CFLAGS = $(WERROR) $(SNMP_CFLAGS) -std=gnu99
 ripd_ripd_snmp_la_LDFLAGS = -avoid-version -module -shared -export-dynamic
 ripd_ripd_snmp_la_LIBADD = lib/libfrrsnmp.la
 

--- a/zebra/subdir.am
+++ b/zebra/subdir.am
@@ -127,7 +127,7 @@ zebra_zebra_irdp_la_SOURCES = \
 zebra_zebra_irdp_la_LDFLAGS = -avoid-version -module -shared -export-dynamic
 
 zebra_zebra_snmp_la_SOURCES = zebra/zebra_snmp.c
-zebra_zebra_snmp_la_CFLAGS = $(WERROR) $(SNMP_CFLAGS)
+zebra_zebra_snmp_la_CFLAGS = $(WERROR) $(SNMP_CFLAGS) -std=gnu99
 zebra_zebra_snmp_la_LDFLAGS = -avoid-version -module -shared -export-dynamic
 zebra_zebra_snmp_la_LIBADD = lib/libfrrsnmp.la
 

